pkg/detectors/dockerhub/dockerhub_test.go

```diff
@@ -0,0 +1,137 @@
+//go:build detectors
+// +build detectors
+
+package dockerhub
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/kylelemons/godebug/pretty"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+func TestDockerhub_FromChunk(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	defer cancel()
+	testSecrets, err := common.GetSecret(ctx, "trufflehog-testing", "detectors2")
+	if err != nil {
+		t.Fatalf("could not get test secrets from GCP: %s", err)
+	}
+	username := testSecrets.MustGetField("DOCKERHUB_USERNAME")
+	email := testSecrets.MustGetField("DOCKERHUB_EMAIL")
+	password := testSecrets.MustGetField("DOCKERHUB_PASSWORD")
+	inactive_password := testSecrets.MustGetField("DOCKERHUB_INACTIVE_PASSWORD")
+
+	type args struct {
+		ctx    context.Context
+		data   []byte
+		verify bool
+	}
+	tests := []struct {
+		name    string
+		s       Scanner
+		args    args
+		want    []detectors.Result
+		wantErr bool
+	}{
+		{
+			name: "found, verified",
+			s:    Scanner{},
+			args: args{
+				ctx:    context.Background(),
+				data:   []byte(fmt.Sprintf("docker login -u %s -p %s", username, password)),
+				verify: true,
+			},
+			want: []detectors.Result{
+				{
+					DetectorType: detectorspb.DetectorType_Dockerhub,
+					Verified:     true,
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "found, verified (email)",
```

Comment:
question: Were you able to get this to also work? I tried to use an email instead of username and got a 401 from the API. Looking at the docs, they only mention username. https://docs.docker.com/docker-hub/api/latest/#tag/authentication/operation/PostUsersLogin 

It's possible their docs aren't updated, and I setup my test incorrectly, so wanted to double check.

Reply:
I don't remember where I found this information 🧐 ... using email works when I send a request via Postman, but it doesn't seem to work on command line. 

Also, I realized the regex was not working as intended which caused the email test to fail, but it should pass now.